test/functional/tests/cache_ops/test_cleaning_policy_operation.py

```diff
@@ -0,0 +1,285 @@
+        time.sleep(time_to_wait)
+
+    with TestRun.step("Check write statistics for core device"):
+        core_writes_after_wait_for_cleaning = (
+            cache.get_statistics().block_stats.core.writes
+        )
+        check_cleaning_policy_operation(
+            cleaning_policy,
+            core_writes_before_wait_for_cleaning,
+            core_writes_after_wait_for_cleaning,
+        )
+
+    with TestRun.step("Stop all caches"):
+        casadm.stop_all_caches()
+        Udev.enable()
+
+
+def storage_prepare():
+    cache_dev = TestRun.disks["cache"]
+    cache_dev.create_partitions([Size(1, Unit.GibiByte)])
+    core_dev = TestRun.disks["core"]
+    parts = [Size(2, Unit.GibiByte)] * cores_count
+    core_dev.create_partitions(parts)
+    return cache_dev, core_dev
+
+
+def set_cleaning_policy_and_params(cache, cleaning_policy):
+    if cleaning_policy != CleaningPolicy.DEFAULT:
+        cache.set_cleaning_policy(cleaning_policy)
+    current_cleaning_policy = cache.get_cleaning_policy()
+    if current_cleaning_policy != cleaning_policy:
+        TestRun.fail(
+            f"Cleaning policy is {current_cleaning_policy}, "
+            f"should be {cleaning_policy}"
+        )
+
+    if cleaning_policy == CleaningPolicy.alru:
+        alru_params = FlushParametersAlru()
+        alru_params.wake_up_time = Time(seconds=10)
+        alru_params.staleness_time = Time(seconds=2)
+        alru_params.flush_max_buffers = 100
+        alru_params.activity_threshold = Time(milliseconds=1000)
+        cache.set_params_alru(alru_params)
+        current_alru_params = cache.get_flush_parameters_alru()
+        if current_alru_params != alru_params:
+            failed_params = ""
+            if current_alru_params.wake_up_time != alru_params.wake_up_time:
+                failed_params += (
+                    f"Wake Up time is {current_alru_params.wake_up_time}, "
+                    f"should be {alru_params.wake_up_time}\n"
+                )
+            if current_alru_params.staleness_time != alru_params.staleness_time:
+                failed_params += (
+                    f"Staleness Time is {current_alru_params.staleness_time}, "
+                    f"should be {alru_params.staleness_time}\n"
+                )
+            if current_alru_params.flush_max_buffers != alru_params.flush_max_buffers:
+                failed_params += (
+                    f"Flush Max Buffers is {current_alru_params.flush_max_buffers}, "
+                    f"should be {alru_params.flush_max_buffers}\n"
+                )
+            if current_alru_params.activity_threshold != alru_params.activity_threshold:
+                failed_params += (
+                    f"Activity Threshold is {current_alru_params.activity_threshold}, "
+                    f"should be {alru_params.activity_threshold}\n"
+                )
+            TestRun.fail(f"ALRU parameters did not switch properly:\n{failed_params}")
```

Comment:
I'm not sure if we should stop the test when it fails (and TestRun.fail does that) - consider using TestRun.LOGGER.error (in the whole file)

Reply:
I believe, that in test step called "Check for running CAS cleaner" there should be TestRun.failed as it is, but like Daniel said, in other steps you can change it to the TestRun.LOGGER.error :)

Reply:
Maybe Kasia is right - I thought about situation, when for some reason cleaner process name changes (then this check would abort test), but after giving it some thought I think that unexpected name could be a test-stopper ;)

Reply:
Done